rest-client/src/test/java/org/sagebionetworks/bridge/rest/UserSessionInterceptorTest.java

```diff
@@ -34,109 +40,288 @@
+
+        try {
+            interceptor.intercept(chain);
+            fail();
+        } catch (Throwable e) {
+            e.printStackTrace();
+            assertTrue(e instanceof EntityNotFoundException);
+
+        }
+
+        assertNull(interceptor.getSession(signIn));
+    }
+
     @Test
     public void interceptsSignOut() throws IOException {
         interceptor.addSession(signIn, userSessionInfo);
         doReturn(request).when(chain).request();
         doReturn(httpUrl).when(request).url();
-        doReturn(Lists.newArrayList("v3","auth","signOut")).when(httpUrl).pathSegments();
+        doReturn(Lists.newArrayList("v3", "auth", "signOut")).when(httpUrl).pathSegments();
         doReturn(response).when(chain).proceed(request);
         doReturn(200).when(response).code();
         doReturn("sessionToken").when(request).header("Bridge-Session");
-        
+
         interceptor.intercept(chain);
-        
+
         assertNull(interceptor.getSession(signIn));
     }
-    
+
     @Test
     public void interceptsSignIn() throws IOException {
         doReturn(response).when(chain).proceed(request);
         doReturn(request).when(chain).request();
-        
+
         // Mock request
         doReturn(httpUrl).when(request).url();
-        doReturn(Lists.newArrayList("v3","auth","signIn")).when(httpUrl).pathSegments();
+        doReturn(Lists.newArrayList("v3", "auth", "signIn")).when(httpUrl).pathSegments();
         doReturn(requestBuilder).when(request).newBuilder();
         doReturn(request).when(requestBuilder).build();
         doReturn(requestBody).when(request).body();
-        doReturn(Tests.unescapeJson("{'study':'studyId','email':'email@email.com','password':'P4ssword'}")).when(buffer).readUtf8();
+        doReturn(Tests.unescapeJson("{'study':'studyId','email':'email@email.com'," +
+                "'password':'P4ssword'}")).when(buffer).readUtf8();
         // Just assume creating a buffer works.
         doReturn(buffer).when(interceptor).createBuffer();
-        
+
         // Mock response
         doReturn(responseBody).when(response).body();
         doReturn(MediaType.parse("application/json")).when(responseBody).contentType();
-        doReturn(Tests.unescapeJson("{'email':'emailInSession@email.com'}")).when(responseBody).string();
+        doReturn(Tests.unescapeJson("{'email':'emailInSession@email.com'}")).when(responseBody)
+                .string();
         // Just assume copying works
         doReturn(response).when(interceptor).copyResponse(any(Response.class), any(String.class));
-        
+
         interceptor.intercept(chain);
-        
+
         SignIn signIn = new SignIn().study("studyId").email("email@email.com").password("P4ssword");
-        
+
         UserSessionInfo info = interceptor.getSession(signIn);
         assertEquals("emailInSession@email.com", info.getEmail());
     }
-    
+
+    @Test
+    public void interceptsNonUserSessionResponses() throws IOException {
```

Comment:
Thanks. this is a dupe of interceptsNonAuthCall_NotUserSessionResponses
sanity checking the interception doesn't barf if it's not signIn, signOut, self/update